From 6c54b8b0b0e1d18b5d75b7ae0ce9439ab0e199bf Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Tue, 12 Dec 2023 17:14:00 +0100
Subject: [PATCH 0719/1141] media: verisilicon: h264: improve encoder
 robustness

Remove H1_REG_ENC_REC_WRITE_BUFFER_4MB bit from H1_REG_ENC_CTRL
in order to limit to 1 macro block the buffered size burst when
writing reconstructed image.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>
Change-Id: If49a81f396a619efd034412e46721bbc32b2978d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/346098
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Domain-Review: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Tested-by: Hugues FRUCHET <hugues.fruchet@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/media/platform/verisilicon/hantro_h1_h264_enc.c | 1 -
 1 file changed, 1 deletion(-)

--- a/drivers/media/platform/verisilicon/hantro_h1_h264_enc.c
+++ b/drivers/media/platform/verisilicon/hantro_h1_h264_enc.c
@@ -417,7 +417,6 @@ int hantro_h1_h264_enc_run(struct hantro
 		| H1_REG_ENC_CTRL_NAL_MODE_BIT
 		| H1_REG_ENC_CTRL_WIDTH(MB_WIDTH(ctx->src_fmt.width))
 		| H1_REG_ENC_CTRL_HEIGHT(MB_HEIGHT(ctx->src_fmt.height))
-		| H1_REG_ENC_REC_WRITE_BUFFER_4MB
 		| H1_REG_ENC_CTRL_ENC_MODE_H264
 		| H1_REG_ENC_CTRL_EN_BIT;
 	if (params->slice_type) {
